whisper.cpp

```diff
@@ -1078,6 +1078,11 @@ static ggml_backend_t whisper_backend_init(const whisper_context_params & params
         if (!backend_gpu) {
             WHISPER_LOG_ERROR("%s: ggml_backend_metal_init() failed\n", __func__);
         }
+        if (!ggml_backend_metal_supports_family(backend_gpu, 7)) {
```

Comment:
I don't think this is correct. It should disable it on family 7 **and older**.

Reply:
Older would be Apple 6, 5, etc.

Reply:
Ah sorry, yes.

I thought somebody said that Apple7 works?

Reply:
Yes, should be disabled on family 6 and older. I got confused since it's disabling 7 here.

Reply:
Yup, let me know if you still think this needs to be changed.

The way it is written now it should do the following:

- Apple 1-6 -> fallback to CPU
- Apple 7, 8, 9, .. -> use GPU

The assumption is that if Apple7 is supported, then due to backwards compatibility all previous families (1-6) are also supported, so no need to check for those individually.

Reply:
the order seems to be wrong, we should check the compatibility first, then call ggml_backend_metal_init. otherwise it still crash on lower device.

Reply:
I think, we should do something like this:
```c
#ifdef GGML_USE_METAL
    if (!ggml_backend_metal_supports_family(backend_gpu, 7)) {
        WHISPER_LOG_ERROR("%s: Metal GPU does not support family 7 - falling back to CPU\n", __func__);
    }
    
    if (params.use_gpu) {
        WHISPER_LOG_INFO("%s: using Metal backend\n", __func__);
        ggml_metal_log_set_callback(whisper_log_callback_default, nullptr);
        backend_gpu = ggml_backend_metal_init();
        if (!backend_gpu) {
            WHISPER_LOG_ERROR("%s: ggml_backend_metal_init() failed\n", __func__);
        }
    }
#endif
```

Reply:
I think we should do something like this:

```c
    if (!ggml_backend_metal_supports_family(backend_gpu, 7)) {
        WHISPER_LOG_ERROR("%s: Metal GPU does not support family 7 - falling back to CPU\n", __func__);
    }
    
    if (params.use_gpu) {
        WHISPER_LOG_INFO("%s: using Metal backend\n", __func__);
        ggml_metal_log_set_callback(whisper_log_callback_default, nullptr);
        backend_gpu = ggml_backend_metal_init();
        if (!backend_gpu) {
            WHISPER_LOG_ERROR("%s: ggml_backend_metal_init() failed\n", __func__);
        }
    }
```

but right now, it's not possible, because we can't get the ggml_backend_t if it is not inited, I think there need some refactor

Reply:
Was this done?

Reply:
This seems done, right now it do check in ggml_backend_metal_init to avoid crash